### DISCUSSION TITLE
How to implement Android's Health Connect in Maui?

### DISCUSSION THREAD

**Discussion by CMTEST080** [dotnet/maui#18860] [Q&A]
.NET MAUI can retrieve healthcare data from iOS's HealthKit, but it cannot retrieve data from Android's Health Connect.

Is there a way to retrieve data from Health Connect using .NET MAUI?

Flutter has a package that can retrieve data from Health Connect. 
https://pub.dev/packages/health

Is it possible to achieve something similar with .NET MAUI

**Comment by stevehawkins** [comment]
Any updates on this? the fit api is closed for signup and discontinued in 2025? surely more people need this as well.

**Comment by nll** [comment]
It seems that there was some work done to create a binding https://github.com/xamarin/AndroidX/issues/825 and a nuget package already exists https://www.nuget.org/packages/Xamarin.AndroidX.Health.Connect.ConnectClient/1.1.0-alpha07

**Comment by osyyyS** [comment]
Anything new here? I need to retrieve data from Health Connect since I cannot sign up for the fit api, do I have to move away from .NET MAUI?

**Reply by JosefPRajmon** [reply]
I know load data, get permion but didn't know how open health seting on dewice with integrade health connect. Do you want code?

**Reply by Blanquitoh** [reply]
> I know load data, get permion but didn't know how open health seting on dewice with integrade health connect. Do you want code?

I'd love to get that code, please.

**Reply by JosefPRajmon** [reply]
Then here is first wersion. In my projekt is  this code refactoried, but this work... its have loading and saving data. I don't need update or remove but i guess that is some mechanism... https://github.com/JosefPRajmon/HealthConnectLibraly

**Reply by JosefPRajmon** [reply]
is it usefull for you?

**Reply by Kebechet** [reply]
Seems that @osyyyS has found our lib sufficient for his usecase
https://github.com/Kebechet/Maui.Health

😇

**Comment by Kebechet** [comment]
You can use our MAUI abstraction wrapping both Android and iOS
https://github.com/Kebechet/Maui.Health
